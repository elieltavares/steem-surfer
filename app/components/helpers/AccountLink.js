import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getAccount } from '../../backend/steem-client';

export const makePath = username => `/@${username}`;

class AccountLink extends Component {
  goProfile = async () => {
    const { username, history, actions, onClick, afterClick } = this.props;

    onClick();

    let { accountData } = this.props;
    const { setVisitingAccount } = actions;

    if (!accountData) {
      try {
        accountData = await getAccount(username);
      } catch (err) {
        accountData = null;
      }
    }

    if (accountData) {
      setVisitingAccount(accountData);
    }

    history.push(makePath(username));

    afterClick();
  };

  render() {
    const { children } = this.props;

    return React.cloneElement(children, {
      onClick: this.goProfile
    });
  }
}

AccountLink.defaultProps = {
  accountData: null,
  onClick: () => {},
  afterClick: () => {}
};

AccountLink.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  accountData: PropTypes.instanceOf(Object),
  onClick: PropTypes.func,
  afterClick: PropTypes.func,
  actions: PropTypes.shape({
    setVisitingAccount: PropTypes.func.isRequired
  }).isRequired
};

export default AccountLink;
