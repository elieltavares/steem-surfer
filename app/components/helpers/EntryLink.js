import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { getContent } from '../../backend/steem-client';

export const makePath = (category, author, permlink, toReplies = false) =>
  `/${category}/@${author}/${permlink}${toReplies ? '#replies' : ''}`;

class EntryLink extends Component {
  goEntry = async () => {
    let { entry } = this.props;
    const { history, actions, toReplies, markAsRead } = this.props;

    if (markAsRead) {
      markAsRead();
    }

    if (!entry) {
      const { author, permlink } = this.props;
      entry = await getContent(author, permlink);
    }

    if (entry) {
      const { category, author, permlink } = entry;
      actions.setVisitingEntry(entry);
      history.push(makePath(category, author, permlink, toReplies));
    }
  };

  render() {
    const { children } = this.props;

    return React.cloneElement(children, {
      onClick: this.goEntry
    });
  }
}

EntryLink.defaultProps = {
  entry: null,
  toReplies: false,
  markAsRead: null
};

EntryLink.propTypes = {
  children: PropTypes.element.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  entry: PropTypes.shape({
    category: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    permlink: PropTypes.string.isRequired
  }),
  author: PropTypes.string.isRequired,
  permlink: PropTypes.string.isRequired,
  toReplies: PropTypes.bool,
  actions: PropTypes.shape({
    setVisitingEntry: PropTypes.func.isRequired
  }).isRequired,
  markAsRead: PropTypes.func
};

export default EntryLink;
