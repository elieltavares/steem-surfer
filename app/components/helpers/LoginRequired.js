import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'antd';
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';

import Login from '../dialogs/Login';

export const checkLogin = (activeAccount, requiredKeys) => {
  // Steem connect login.
  if (activeAccount && activeAccount.type === 'sc') {
    return true;
  }

  // Traditional login. Check all required keys exists in user keys.
  if (
    activeAccount &&
    activeAccount.type === 's' &&
    requiredKeys.every(e => activeAccount.keys[e])
  ) {
    return true;
  }

  return false;
};

class LoginRequired extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loginModalVisible: false
    };
  }

  showLoginModal = () => {
    this.setState({
      loginModalVisible: true
    });

    setTimeout(() => {
      const { onDialogOpen } = this.props;
      onDialogOpen();
    }, 200);
  };

  onLoginModalCancel = () => {
    this.setState({
      loginModalVisible: false
    });
  };

  onLoginSuccess = () => {
    this.setState({
      loginModalVisible: false
    });

    setTimeout(() => {
      if (!this.check()) {
        this.showLoginModal();
      }
    }, 400);
  };

  afterModalClose = () => {
    const { onDialogClose } = this.props;
    onDialogClose();
  };

  check = () => {
    const { activeAccount, requiredKeys } = this.props;

    return checkLogin(activeAccount, requiredKeys);
  };

  render() {
    const { children } = this.props;

    if (this.check()) {
      return children;
    }

    const { activeAccount, requiredKeys } = this.props;
    const { loginModalVisible } = this.state;

    // Default message
    let loginMsg = <FormattedMessage id="login-required.default" />;

    if (activeAccount) {
      // More specific, key based message
      loginMsg =
        requiredKeys.length > 1 ? (
          <FormattedHTMLMessage
            id="login-required.keys-required"
            values={{ keys: requiredKeys }}
          />
        ) : (
          <FormattedHTMLMessage
            id="login-required.key-required"
            values={{ key: requiredKeys }}
          />
        );
    }

    // Clone child element rewriting onclick event
    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.showLoginModal();
      }
    });

    return (
      <Fragment>
        {newChild}

        <Modal
          visible={loginModalVisible}
          onCancel={this.onLoginModalCancel}
          footer={false}
          width="500px"
          closable={false}
          destroyOnClose
          centered
          afterClose={this.afterModalClose}
        >
          <Login
            {...this.props}
            onSuccess={this.onLoginSuccess}
            loginMsg={loginMsg}
          />
        </Modal>
      </Fragment>
    );
  }
}

LoginRequired.defaultProps = {
  activeAccount: null,
  onDialogOpen: () => {},
  onDialogClose: () => {},
  requiredKeys: []
};

LoginRequired.propTypes = {
  children: PropTypes.element.isRequired,
  activeAccount: PropTypes.instanceOf(Object),
  requiredKeys: PropTypes.arrayOf(PropTypes.string),
  onDialogOpen: PropTypes.func,
  onDialogClose: PropTypes.func
};

export default LoginRequired;
