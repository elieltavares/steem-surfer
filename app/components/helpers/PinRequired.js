import React, { Component, Fragment } from 'react';
import { Modal } from 'antd';

import PropTypes from 'prop-types';
import PinConfirm from '../dialogs/PinConfirm';

class PinRequired extends Component {
  constructor(props) {
    super(props);

    this.state = {
      show: false
    };
  }

  onPinInvalidated = () => {
    this.setState({ show: false });
  };

  onConfirmPinSuccess = value => {
    const { onSuccess } = this.props;
    this.setState({ show: false }, () => {
      onSuccess(value);
    });
  };

  cancelFn = () => {
    this.setState({ show: false });
  };

  render() {
    const { children } = this.props;
    const { show } = this.state;

    // Clone child element rewriting onclick event
    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.setState({ show: true });
      }
    });

    return (
      <Fragment>
        {show && (
          <Modal
            footer={null}
            closable
            onCancel={this.cancelFn}
            keyboard
            visible
            width="500px"
            centered
            destroyOnClose
          >
            <PinConfirm
              {...this.props}
              onSuccess={this.onConfirmPinSuccess}
              onInvalidate={this.onPinInvalidated}
            />
          </Modal>
        )}
        {newChild}
      </Fragment>
    );
  }
}

PinRequired.defaultProps = {
  onSuccess: () => {}
};

PinRequired.propTypes = {
  children: PropTypes.element.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  onSuccess: PropTypes.func
};

export default PinRequired;
