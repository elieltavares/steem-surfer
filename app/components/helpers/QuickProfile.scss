@import '../../style/mixins';
@import '../../style/colors';

.quick-profile-content {
  height: 100%;
  left: 0;
  padding: 8px 16px;
  position: absolute;
  top: 0;
  width: 100%;

  .profile-area {
    align-items: center;
    cursor: default;
    display: flex;
    flex-direction: column;
    height: 240px;
    justify-content: center;
    position: relative;
    text-align: center;
    width: 100%;

    .account-cover {
      align-items: center;
      background-color: $white-three;
      background-position: 50% 50%;
      background-repeat: no-repeat;
      background-size: cover;
      border-radius: 10px;
      display: flex;
      height: 134px;
      justify-content: center;
      margin-bottom: 16px;
      position: relative;
      width: 100%;

      .profile-avatar {
        height: 100px;
        position: relative;
        width: 100px;

        .user-avatar {
          border: 3px solid $powder-blue;
          height: 100px;
          width: 100px;
        }

        .reputation {
          background: $dark-sky-blue;
          border-radius: 25px;
          bottom: 0;
          color: $white;
          font-weight: 700;
          left: 0;
          line-height: 1em;
          padding: 4px 12px;
          position: absolute;
        }
      }

      .follow-btn-holder {
        position: absolute;
        right: 20px;
        top: 20px;
        z-index: 1;
      }
    }

    .username {
      color: $dark-sky-blue;
      cursor: pointer;
      font-size: 14px;
      font-weight: 700;
      margin-bottom: 12px;

      &:before {
        content: '@';
      }

      &:hover {
        text-decoration: underline;
      }
    }

    .about {
      color: $dark-sky-blue;
      height: 16px;
      margin-bottom: 10px;
      max-width: 90%;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;

      @include themify(day) {
        opacity: 0.6;
      }

      @include themify(night) {
        opacity: 0.8;
      }
    }

    .numbers {
      align-items: center;
      color: $steel-grey;
      display: flex;
      font-size: 13px;
      height: 20px;

      .mi {
        color: $silver;
        font-size: 16px;
        margin-right: 8px;
      }

      .followers,
      .post-count,
      .following {
        align-items: center;
        display: flex;
        height: 100%;
      }

      .followers,
      .post-count {
        margin-right: 24px;
      }

      .post-count {
        .mi {
          font-size: 18px;
        }
      }
    }
  }

  .account-menu {
    display: flex;
    height: 44px;
    margin: 0 6px;

    @include themify(day) {
      border-bottom: 1px solid $white-four;
    }

    @include themify(night) {
      border-bottom: 1px solid $dark-two;
    }

    .account-menu-items {
      display: flex;
      justify-content: space-around;
      width: 100%;

      .menu-item {
        align-items: center;
        color: $steel-grey;
        display: flex;
        font-size: 14px;
        height: 44px;
        padding: 0 16px;

        &:not(:last-child) {
          margin-right: 30px;
        }

        &.selected-item {
          border-bottom: 2px solid $dark-sky-blue;
          color: $dark-sky-blue;
          font-weight: 700;
        }
      }

      .separator-item {
        display: flex;
        flex-grow: 1;
        height: 20px;
      }
    }
  }

  .entries {
    max-height: calc(100% - 240px - 44px);
    overflow: auto;
    margin: 0 0 0 6px;

    .entry-list-item .app,
    .entry-list-loading-item .app {
      display: none;
    }
  }

  &.loading {
    .account-menu {
      border-bottom-color: transparent;

      .account-menu-items {
        .menu-item {
          pointer-events: none;
          border-bottom-color: transparent;
        }
      }
    }
  }
}
