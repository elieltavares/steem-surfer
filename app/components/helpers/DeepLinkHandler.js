import React from 'react';
import PropTypes from 'prop-types';

import filters from '../../constants/filters.json';

import { makePath as makePathTag } from './TagLink';
import { makePath as makePathAccount } from './AccountLink';
import { makePath as makePathEntry } from './EntryLink';

export const deepLink2Obj = url => {
  let urlPart = url.split('://')[1];

  // remove last char if /
  if (urlPart.endsWith('/')) {
    urlPart = urlPart.substring(0, urlPart.length - 1);
  }

  const parts = urlPart.split('/');

  // witnesses
  if (parts.length === 1 && parts[0] === 'witnesses') {
    return { type: 'witnesses' };
  }

  // sps
  if (parts.length === 1 && parts[0] === 'sps') {
    return { type: 'sps' };
  }

  // filter
  if (parts.length === 1 && filters.includes(parts[0])) {
    return { type: 'filter', filter: parts[0] };
  }

  // filter with tag
  if (parts.length === 2 && filters.includes(parts[0])) {
    return { type: 'filter-tag', filter: parts[0], tag: parts[1] };
  }

  // account
  if (parts.length === 1 && parts[0].startsWith('@')) {
    return { type: 'account', account: parts[0].replace('@', '') };
  }

  // post without tag
  if (parts.length === 2 && parts[0].startsWith('@')) {
    return {
      type: 'post',
      cat: 'esteem',
      author: parts[0].replace('@', ''),
      permlink: parts[1]
    };
  }

  // post with tag
  if (parts.length === 3 && parts[1].startsWith('@')) {
    return {
      type: 'post',
      cat: parts[0],
      author: parts[1].replace('@', ''),
      permlink: parts[2]
    };
  }
};

class DeepLinkHandler extends React.Component {
  componentDidMount() {
    window.addEventListener('deep-link', this.handler);
  }

  componentWillUnmount() {
    window.removeEventListener('deep-link', this.handler);
  }

  handler = e => {
    const { history } = this.props;
    const { url } = e.detail;
    const obj = deepLink2Obj(url);

    let path;

    switch (obj.type) {
      case 'filter':
        path = `/${obj.filter}`;
        break;
      case 'filter-tag':
        path = makePathTag(obj.filter, obj.tag);
        break;
      case 'account':
        path = makePathAccount(obj.account);
        break;
      case 'post':
        path = makePathEntry(obj.cat, obj.author, obj.permlink);
        break;
      case 'sps':
        path = `/sps`;
        break;
      case 'witnesses':
        path = `/witnesses`;
        break;
      default:
        path = null;
        break;
    }

    if (path) {
      history.push(path);
    }
  };

  render() {
    return null;
  }
}

DeepLinkHandler.propTypes = {
  history: PropTypes.shape({}).isRequired
};

export default DeepLinkHandler;
