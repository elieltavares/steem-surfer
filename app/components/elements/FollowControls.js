import React, { Component } from 'react';

import PropTypes from 'prop-types';

import { Button, message } from 'antd';
import { FormattedMessage } from 'react-intl';

import LoginRequired from '../helpers/LoginRequired';
import {
  getFollowing,
  follow,
  unFollow,
  ignore
} from '../../backend/steem-client';

import formatChainError from '../../utils/format-chain-error';

class FollowControls extends Component {
  constructor(props) {
    super(props);

    this.state = {
      fetching: true,
      processing: false,
      following: false,
      muted: false
    };
  }

  async componentDidMount() {
    this.mounted = true;

    const { activeAccount } = this.props;

    if (!activeAccount) {
      this.stateSet({ fetching: false });
      return;
    }

    await this.fetchStatus();

    this.stateSet({ fetching: false });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  stateSet = (obj, cb = undefined) => {
    if (this.mounted) {
      this.setState(obj, cb);
    }
  };

  isFollowing = async () => {
    const { activeAccount, targetUsername } = this.props;
    const { username } = activeAccount;

    let resp;
    try {
      resp = await getFollowing(username, targetUsername, 'blog', 1);
    } catch (err) {
      return false;
    }

    if (resp && resp.length > 0) {
      if (
        resp[0].follower === username &&
        resp[0].following === targetUsername
      ) {
        return true;
      }
    }

    return false;
  };

  isMuted = async () => {
    const { activeAccount, targetUsername } = this.props;
    const { username } = activeAccount;

    let resp;
    try {
      resp = await getFollowing(username, targetUsername, 'ignore', 1);
    } catch (err) {
      return false;
    }

    if (resp && resp.length > 0) {
      if (
        resp[0].follower === username &&
        resp[0].following === targetUsername
      ) {
        return true;
      }
    }

    return false;
  };

  fetchStatus = async () => {
    const following = await this.isFollowing();

    // No need to check if muted when already following
    const muted = following ? false : await this.isMuted();

    this.stateSet({ following, muted });
  };

  follow = async () => {
    const { activeAccount, targetUsername, global } = this.props;
    const { pin } = global;

    this.stateSet({ processing: true });
    try {
      await follow(activeAccount, pin, targetUsername);
      this.stateSet({ following: true, muted: false });
    } catch (err) {
      message.error(formatChainError(err));
    } finally {
      this.stateSet({ processing: false });
    }
  };

  unFollow = async () => {
    const { activeAccount, targetUsername, global } = this.props;
    const { pin } = global;

    this.stateSet({ processing: true });
    try {
      await unFollow(activeAccount, pin, targetUsername);
      this.stateSet({ following: false, muted: false });
    } catch (err) {
      message.error(formatChainError(err));
    } finally {
      this.stateSet({ processing: false });
    }
  };

  mute = async () => {
    const { activeAccount, targetUsername, global } = this.props;
    const { pin } = global;

    this.stateSet({ processing: true });
    try {
      await ignore(activeAccount, pin, targetUsername);
      this.stateSet({ following: false, muted: true });
    } catch (err) {
      message.error(formatChainError(err));
    } finally {
      this.stateSet({ processing: false });
    }
  };

  render() {
    const { following, muted, fetching, processing } = this.state;

    const followMsg = <FormattedMessage id="follow-controls.follow" />;
    const unFollowMsg = <FormattedMessage id="follow-controls.unFollow" />;
    const muteMsg = <FormattedMessage id="follow-controls.mute" />;
    const unMuteMsg = <FormattedMessage id="follow-controls.unMute" />;

    const btnFollow = (
      <LoginRequired {...this.props} requiredKeys={['posting']}>
        <Button
          type="primary"
          style={{ marginRight: '5px' }}
          disabled={processing}
          onClick={this.follow}
        >
          {followMsg}
        </Button>
      </LoginRequired>
    );

    const btnUnfollow = (
      <LoginRequired {...this.props} requiredKeys={['posting']}>
        <Button
          type="primary"
          style={{ marginRight: '5px' }}
          disabled={processing}
          onClick={this.unFollow}
        >
          {unFollowMsg}
        </Button>
      </LoginRequired>
    );

    const btnMute = (
      <LoginRequired {...this.props} requiredKeys={['posting']}>
        <Button disabled={processing} onClick={this.mute}>
          {muteMsg}
        </Button>
      </LoginRequired>
    );

    const btnUnMute = (
      <LoginRequired {...this.props} requiredKeys={['posting']}>
        <Button disabled={processing} onClick={this.unFollow}>
          {unMuteMsg}
        </Button>
      </LoginRequired>
    );

    if (fetching) {
      return (
        <div>
          <Button type="primary" disabled style={{ marginRight: '5px' }}>
            {followMsg}
          </Button>
          <Button disabled>{muteMsg}</Button>
        </div>
      );
    }

    if (following) {
      return (
        <div>
          {btnUnfollow}
          {btnMute}
        </div>
      );
    }

    if (muted) {
      return (
        <div>
          {btnFollow}
          {btnUnMute}
        </div>
      );
    }

    return (
      <div>
        {btnFollow}
        {btnMute}
      </div>
    );
  }
}

FollowControls.defaultProps = {
  activeAccount: null
};

FollowControls.propTypes = {
  global: PropTypes.shape({
    pin: PropTypes.string
  }).isRequired,
  activeAccount: PropTypes.instanceOf(Object),
  targetUsername: PropTypes.string.isRequired
};

export default FollowControls;
