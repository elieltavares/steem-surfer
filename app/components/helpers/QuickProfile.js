/*
eslint-disable import/no-cycle
*/

import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Drawer, message } from 'antd';
import { FormattedMessage, FormattedNumber } from 'react-intl';

import { proxifyImageSrc } from '@esteemapp/esteem-render-helpers';

import Tooltip from '../common/Tooltip';

import {
  getAccount,
  getDiscussions,
  getFollowCount
} from '../../backend/steem-client';
import authorReputation from '../../utils/author-reputation';

import UserAvatar from '../elements/UserAvatar';
import EntryListLoadingItem from '../elements/EntryListLoadingItem';
import LinearProgress from '../common/LinearProgress';
import EntryListItem from '../elements/EntryListItem';
import FollowControls from '../elements/FollowControls';
import AccountLink from './AccountLink';
import coverFallbackDay from '../../img/cover-fallback-day.png';
import coverFallbackNight from '../../img/cover-fallback-night.png';

class QuickProfile extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false,
      active: false,
      profile: {
        name: ' ',
        about: ' ',
        coverImage: null,
        postCount: 0
      },
      follows: {
        followerCount: null,
        followingCount: null
      },
      loadingEntries: true,
      entries: [],
      accountData: null
    };
  }

  load = async () => {
    const { username, intl } = this.props;

    const account = await getAccount(username);

    this.setState({ accountData: account });

    let accountProfile;
    try {
      accountProfile = JSON.parse(account.json_metadata).profile;
    } catch (err) {
      accountProfile = {};
    }

    if (accountProfile) {
      const name = accountProfile.name || null;
      const about = accountProfile.about || null;
      const postCount = account.post_count;
      const coverImage = accountProfile.cover_image || null;

      this.setState({ profile: { name, about, coverImage, postCount } });
    }

    let follow;
    try {
      follow = await getFollowCount(username);
    } catch (err) {
      message.error(
        intl.formatMessage({ id: 'quick-profile.fetch-error-profile' })
      );
    }

    if (follow) {
      const followerCount = follow.follower_count;
      const followingCount = follow.following_count;

      this.setState({ follows: { followerCount, followingCount } });
    }

    let entries;
    try {
      entries = await getDiscussions('blog', {
        tag: username,
        limit: 7,
        start_author: undefined,
        start_permlink: undefined
      });
    } catch (err) {
      message.error(
        intl.formatMessage({ id: 'quick-profile.fetch-error-content' })
      );
    }

    if (entries) {
      this.setState({ entries, loadingEntries: false });
    }
  };

  show = () => {
    this.setState({ active: true, visible: true });
    this.load();
  };

  hide = () => {
    this.setState({ visible: false });
    setTimeout(() => {
      this.setState({ active: false });
    }, 500);
  };

  goSection = section => {
    const { history, username } = this.props;
    const u = `/@${username}/${section}`;
    history.push(u);
  };

  render() {
    const { children, username, global, intl } = this.props;
    let { reputation } = this.props;
    const {
      visible,
      active,
      profile,
      follows,
      entries,
      loadingEntries,
      accountData
    } = this.state;

    if (accountData) ({ reputation } = accountData);

    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.show();
      }
    });

    const followersMsg =
      follows.followerCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={follows.followerCount} />
      );
    const postCountMsg =
      profile.postCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={profile.postCount} />
      );
    const followingMsg =
      follows.followingCount === null ? (
        '--'
      ) : (
        <FormattedNumber value={follows.followingCount} />
      );

    const bgImage =
      (profile.coverImage && proxifyImageSrc(profile.coverImage)) ||
      (global.theme === 'day' ? coverFallbackDay : coverFallbackNight);

    return (
      <Fragment>
        {newChild}

        {active && (
          <Drawer
            placement="right"
            closable={false}
            onClose={this.hide}
            visible={visible}
            width="640px"
          >
            <div
              className={`quick-profile-content ${
                loadingEntries ? 'loading' : ''
              } `}
            >
              <div className="profile-area">
                <div
                  className="account-cover"
                  style={{ backgroundImage: `url('${bgImage}')` }}
                >
                  <div className="profile-avatar">
                    <UserAvatar {...this.props} user={username} size="large" />
                    <div className="reputation">
                      {authorReputation(reputation)}
                    </div>
                  </div>
                  <div className="follow-btn-holder">
                    <FollowControls {...this.props} targetUsername={username} />
                  </div>
                </div>
                <AccountLink
                  {...this.props}
                  username={username}
                  accountData={accountData}
                >
                  <div className="username">{username}</div>
                </AccountLink>
                <div className="about">{profile.about}</div>
                <div className="numbers">
                  <Tooltip
                    title={intl.formatMessage({
                      id: 'quick-profile.post-count'
                    })}
                  >
                    <span className="post-count">
                      <i className="mi">list</i>
                      {postCountMsg}
                    </span>
                  </Tooltip>
                  <Tooltip
                    title={intl.formatMessage({
                      id: 'quick-profile.followers'
                    })}
                  >
                    <span className="followers">
                      <i className="mi">people</i>
                      {followersMsg}
                    </span>
                  </Tooltip>
                  <Tooltip
                    title={intl.formatMessage({
                      id: 'quick-profile.following'
                    })}
                  >
                    <span className="following">
                      <i className="mi">person_add</i>
                      {followingMsg}
                    </span>
                  </Tooltip>
                </div>
              </div>

              <div className="account-menu">
                <div className="account-menu-items">
                  <a
                    role="none"
                    className="menu-item selected-item"
                    onClick={() => {
                      this.goSection('blog');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-blog" />
                  </a>
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('comments');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-comments" />
                  </a>
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('replies');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-replies" />
                  </a>
                  <span className="separator-item" />
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('points');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-points" />
                  </a>
                  <a
                    role="none"
                    className="menu-item"
                    onClick={() => {
                      this.goSection('wallet');
                    }}
                  >
                    <FormattedMessage id="quick-profile.section-wallet" />
                  </a>
                </div>
              </div>

              {loadingEntries && (
                <Fragment>
                  <LinearProgress />
                  <div className="entries">
                    <EntryListLoadingItem />
                  </div>
                </Fragment>
              )}

              {!loadingEntries && (
                <div className="entries">
                  {entries.map(d => (
                    <EntryListItem
                      key={`${d.author}-${d.permlink}`}
                      {...this.props}
                      entry={d}
                      asAuthor={username}
                      inDrawer
                    />
                  ))}
                </div>
              )}
            </div>
          </Drawer>
        )}
      </Fragment>
    );
  }
}

QuickProfile.defaultProps = {};

QuickProfile.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  reputation: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  global: PropTypes.shape({
    theme: PropTypes.string.isRequired
  }).isRequired,
  intl: PropTypes.instanceOf(Object).isRequired
};

export default QuickProfile;
