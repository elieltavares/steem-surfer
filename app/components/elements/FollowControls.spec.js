/* eslint-disable */

import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FollowControls from './FollowControls';
import { mountWithIntl, shallowWithIntl } from '../../helpers/intl-test';

Enzyme.configure({ adapter: new Adapter() });

describe('FollowControls', () => {
  it('(1) Default render. No active user.', () => {
    const props = {
      global: { pin: '1q2w' },
      activeAccount: null,
      targetUsername: 'bar'
    };
    const wrapper = mountWithIntl(<FollowControls {...props} />);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('(2) Render with active account', () => {
    const props = {
      global: { pin: '1q2w' },
      activeAccount: { username: 'fooo1' },
      targetUsername: 'bar11'
    };
    const wrapper = mountWithIntl(<FollowControls {...props} />);

    // No following and muted
    wrapper.setState({ fetching: false, following: false, muted: false });
    wrapper.update();

    expect(wrapper.html()).toMatchSnapshot();

    // Following
    wrapper.setState({ fetching: false, following: true, muted: false });
    wrapper.update();

    expect(wrapper.html()).toMatchSnapshot();

    // Muted
    wrapper.setState({ fetching: false, following: false, muted: true });
    wrapper.update();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
