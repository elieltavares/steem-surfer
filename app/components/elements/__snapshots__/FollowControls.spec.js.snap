// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowControls (1) Default render. No active user. 1`] = `"<div><button style=\\"margin-right: 5px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Follow</span></button><button type=\\"button\\" class=\\"ant-btn\\"><span>Mute</span></button></div>"`;

exports[`FollowControls (2) Render with active account 1`] = `"<div><button style=\\"margin-right: 5px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Follow</span></button><button type=\\"button\\" class=\\"ant-btn\\"><span>Mute</span></button></div>"`;

exports[`FollowControls (2) Render with active account 2`] = `"<div><button style=\\"margin-right: 5px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Unfollow</span></button><button type=\\"button\\" class=\\"ant-btn\\"><span>Mute</span></button></div>"`;

exports[`FollowControls (2) Render with active account 3`] = `"<div><button style=\\"margin-right: 5px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Follow</span></button><button type=\\"button\\" class=\\"ant-btn\\"><span>Unmute</span></button></div>"`;
